### ISSUE TITLE
Switch to an fft library in npm

### ISSUE THREAD

**Issue by hughrawlinson** [hughrawlinson/meyda#138]
We're using an fft library as a git include. It's a bit wonky... We should switch to using one from npm, it looks like there are several, including one from the excellent @jensnockert.

**Comment by jensnockert** [comment]
Mine is undermaintained for now, and I wouldn't recommend it :c

**Comment by hughrawlinson** [comment]
Ah ok, thanks Jens.

Still there are a few on npm that don't cause our new linter to have a heart attack, we should move to those :)

**Comment by nevosegal** [comment]
I would love to write an FFT library myself. I'll try making some time for it soon

**Comment by hughrawlinson** [comment]
@nevosegal would you write it as a module or as part of Meyda? Are there advantages to either?

**Comment by nevosegal** [comment]
@hughrawlinson as a separate module. I don't think there are any advantages to either really, it's just that I would like people to use it so it's probably easier that way.

**Comment by nevosegal** [comment]
Quick update - FFT library underway! available [here](https://github.com/nevosegal/fft-js).

**Comment by hughrawlinson** [comment]
@nevosegal is your FFT library at feature parity with our existing dependency and in NPM now? Looks like it still needs tests and a README, but there are no issues, so I don't know where to start helping.

**Comment by nevosegal** [comment]
@hughrawlinson it's not yet on npm, and it does indeed need some tests before we can integrate it with Meyda. Hopefully I'll be able to do it soon.

**Comment by Cortexelus** [comment]
Should compare any FFT  implementation to the other javascript benchmarks. https://www.google.com/amp/s/thebreakfastpost

Speed is essential for online analysis. Phone browsers may struggle if too slow.

**Comment by hughrawlinson** [comment]
Hey @Cortexelus - is that the link you meant to post? It just takes me to a Google redirect for 'thebreakfastpost'.

Do you have any fft benchmarking resources we could refer to? Of course, feel free to send a Pull Request! 😉 I seem to remember @notthetup did something along those lines a couple of years ago, maybe that's still up to date.

**Comment by notthetup** [comment]
https://github.com/notthetup/webaudiofftperf

But I haven't been keeping it up either. :(

 I'll try to at least move up to the latest versions of the fft libraries over this weekend.

**Comment by hughrawlinson** [comment]
Thanks @notthetup! I'll subscribe to the repo to keep up to date 😄

**Comment by 2xAA** [comment]
jsfft is now on npm at 0.0.2 and the git include is currently breaking things.
Files have slightly changed too and ```lib``` is now ```dist```.

Hope to see this updated soon, modV is broken without Meyda.
Might PR tonight if I have time.

**Comment by 2xAA** [comment]
I'm having issues making this PR, uglify is being passed the ComplexArray class from jsfft, babelify isn't transpiling it. I can't find a fix for this currently apart from "it's a bug with Babel 6" on a number of stackoverflow questions.

I know this is another issue entirely and probably should be an issue on a branch of a fork I should make, but just keeping you up to date on this PR status.

**Comment by hughrawlinson** [comment]
Oh crap, only just saw this! Sorry @2xAA!

**Comment by hughrawlinson** [comment]
@nevosegal I don't suppose we can just swap in your FFT implementation?

@2xAA if you have code already, it'd be great if you could open a PR even if it's broken, that way we can see what you've been doing and work together?

**Comment by 2xAA** [comment]
PR #181 now up for review, buildWeb still fails though :(

**Comment by 2xAA** [comment]
So, I switched out the build process for webpack 1 and Babili with the fixes from #181 included and meyda builds again. (I really couldn't figure out why Uglify didn't work. I was at it for hours)

I know @nevosegal is going to incorporate his fft library into meyda soon, but if you're in need of a quick fix you can check my fork out: https://github.com/2xAA/meyda/tree/webpack-build

**Comment by 2xAA** [comment]
#193 potentially solves this

**Comment by hughrawlinson** [comment]
We do this now.
